src/verifier/step3/step3-data-contract-gen.py

```diff
@@ -0,0 +1,101 @@
+import json
+import sys
+import os
+
+def h(s):
+    """
+    Hexify a string
+    """
+    return "0x" + s
+
+def step3DataContractGen(digest, nifs_secondary, r_U_secondary, l_u_secondary):
+    # Unpack nifs_secondary
+    nifs_secondary_repr = nifs_secondary["comm_T"]["comm"]["repr"]
+    output = f"\t\tuint8[32] memory nifs_array;\n"
+    for idx, val in enumerate(nifs_secondary_repr):
+        output += f"\t\tnifs_array[{idx}] = {val};\n"
+
+    output += "\t\tbytes32 nifs = Field.uint8ArrayToBytes32(nifs_array);\n"
+    # Unpack r_U_secondary
+    comm_W = r_U_secondary["comm_W"]["comm"]
+    output += f"\t\tuint256 r_U_comm_W = {h(comm_W)};\n"
+    comm_E = r_U_secondary["comm_E"]["comm"]
+    output += f"\t\tuint256 r_U_comm_E = {h(comm_E)};\n"
+    r_X = r_U_secondary["X"]
+    output += f"\t\tuint256[] memory r_U_X = new uint256[]({len(r_X)});\n"
+    for idx, val in enumerate(r_X):
+        output += f"\t\tr_U_X[{idx}] = {h(val)};\n"
```

Comment:
I believe that parsing `r_U_x` should correspond to the way how limbs are derived in the implementation